client/components/Order/OrderListComponent.js

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import moment from 'moment';
+
+import { Link } from 'react-router';
+
+import RaisedButton from 'material-ui/RaisedButton';
+import Paper from 'material-ui/Paper';
+import TextField from 'material-ui/TextField';
+import CircularProgress from 'material-ui/CircularProgress';
+
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableHeaderColumn,
+  TableRow,
+  TableRowColumn
+} from 'material-ui/Table';
+
+import { refs } from '../../util/firebase';
+
+class OrderList extends React.Component {
+  // static propTypes = {
+  //   viewer: React.PropTypes.object.isRequired
+  // };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      tempOrders: [],
+      orders: [],
+      isSearching: false,
+    };
+  }
+
+  componentDidMount() {
+    refs.order.root.once('value', (data) => {
+      this.setState({
+        tempOrders: Object.keys(data.val()).map(key => data.val()[key])
+      }, () => {
+        this.setState({ orders: this.state.tempOrders }, () => {
+          this.orderRootChildAdded = refs.order.root.orderByKey().on('child_added', (order) => {
+            let isIn = false;
```

Comment:
Because 'child_added' event happens when the case all the items are loaded 'at first' besides the cases they are added, to prevent the former case, i've added logic checking if the added item is in this.state.users.